core/DataAccess/Model.php

```diff
@@ -73,28 +83,116 @@ public function getInvalidatedArchiveIdsSafeToDelete($archiveTable, array $idSit
             $duplicateArchives = explode(',', $row['archives']);
             $countOfArchives = count($duplicateArchives);
 
-            $firstArchive = array_shift($duplicateArchives);
-            list($firstArchiveId, $firstArchiveValue) = explode('.', $firstArchive);
-
             // if there is more than one archive, the older invalidated ones can be deleted
             if ($countOfArchives > 1) {
+                array_shift($duplicateArchives); // we don't want to delete the latest archive if it is usable
+
                 foreach ($duplicateArchives as $pair) {
                     if (strpos($pair, '.') === false) {
                         $this->logger->info("GROUP_CONCAT cut off the query result, you may have to purge archives again.");
                         break;
                     }
 
                     list($idarchive, $value) = explode('.', $pair);
-                    if ($value == ArchiveWriter::DONE_INVALIDATED) {
-                        $archiveIds[] = $idarchive;
-                    }
+                    $archiveIds[] = $idarchive; // does not matter what the value is, the latest is usable so older archives can be purged
                 }
             }
         }
 
         return $archiveIds;
     }
 
+    public function getPlaceholderArchiveIds($archiveTable, array $idSites)
+    {
+        $idSites = array_map('intval', $idSites);
+
+        $sql = "SELECT DISTINCT idarchive FROM `$archiveTable` WHERE ts_archived IS NULL AND idsite IN (" . implode(',', $idSites) . ")";
+        $result = Db::fetchAll($sql);
+        $result = array_column($result, 'idarchive');
+        return $result;
+    }
+
+    public function updateArchiveAsInvalidated($archiveTable, $idSites, $allPeriodsToInvalidate, Segment $segment = null, $forceInvalidateNonexistantRanges = false)
```

Comment:
Explanation part 9: invalidation now creates dummy archives if they don't already exist, so core:archive can pick them up. These archives have a ts_archived of NULL and are meant to be ignored in every other context, except when purging archives. Making this change seemed to fix a couple bugs in the invalidation logic (which can be seen in the changes to the tests).